tools/src/main/java/org/nmdp/ngs/tools/ValidateInterpretation.java

```diff
@@ -122,45 +122,53 @@ public Integer call() throws Exception {
-                    if (shouldValidate(o, observedAlleleList)) {
-
-                        // for each matching sample in expected
-                        for (Interpretation e : expected.get(sample)) {
-                            Genotype expectedGenotype = asGenotype(e);
-                            for (Haplotype expectedHaplotype : expectedGenotype.getHaplotypes()) {
-                                for (AlleleList expectedAlleleList : expectedHaplotype.getAlleleLists()) {
-                                    if (sameLocus(observedAlleleList, expectedAlleleList)) {
-
-                                        // for each pair of expected and observed alleles
-                                        for (Allele expectedAllele : expectedAlleleList.getAlleles()) {
-                                            boolean match = false;
-
-                                            for (Allele observedAllele : observedAlleleList.getAlleles()) {
-                                                if (matchByField(expectedAllele.getGlstring(), observedAllele.getGlstring()) >= resolution) {
-                                                    match = true;
-                                                    break;
-                                                }
-                                            }
-
-                                            if (match) {
-                                                passes++;
-                                            }
-                                            else {
-                                                failures++;
-                                            }
-
-                                            if (!printSummary) {
-                                                writer.println((match ? "PASS" : "FAIL") + "\t" + sample + "\t" + expectedAllele);
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
+            	
+            // for each matching sample in expected
+            	for (Interpretation e : expected.get(sample)) {
+            		Genotype expectedGenotype = asGenotype(e);
+            		for (Haplotype expectedHaplotype : expectedGenotype.getHaplotypes()) {
+            			for (AlleleList expectedAlleleList : expectedHaplotype.getAlleleLists()) {
+            				if (shouldValidate(e, expectedAlleleList)) {
+            					// for each pair of expected and observed alleles
+            					for (Allele expectedAllele : expectedAlleleList.getAlleles()) {
+
+            						boolean match = false;
+
+            						for (Interpretation o : observed.get(sample)) {
+            							AlleleList observedAlleleList = asAlleleList(o);
+
+            							if (sameLocus(observedAlleleList, expectedAlleleList)) {
+            								if (shouldValidate(o, observedAlleleList)) {
```

Comment:
not sure this needs to be called twice (once for expected and once for observed) since loci already match